lib/src/mixins/batched_redraws.dart

```diff
@@ -21,26 +21,30 @@ class _RedrawScheduler implements Function {
 
   Future _tick() async {
     await window.animationFrame;
-    _components
-      ..forEach((component, callbacks) {
-        // Skip if the component doesn't want to batch redraw
-        if (!component.shouldBatchRedraw) {
-          return;
-        }
 
-        var chainedCallbacks;
+    var entries = _components.entries.toList();
```

Comment:
Can we add a comment that we're saving a copy of this so we don't iterate over the map as it's potentially being mutated?